src/js/_enqueues/lib/emoji-loader.js

```diff
@@ -16,23 +38,36 @@
 	 *
 	 * @private
 	 *
+	 * @param {CanvasRenderingContext2D} context 2D Context.
 	 * @param {string} set1 Set of Emoji to test.
 	 * @param {string} set2 Set of Emoji to test.
 	 *
 	 * @return {boolean} True if the two sets render the same.
 	 */
-	function emojiSetsRenderIdentically( set1, set2 ) {
+	function emojiSetsRenderIdentically( context, set1, set2 ) {
 		// Cleanup from previous test.
-		context.clearRect( 0, 0, canvas.width, canvas.height );
+		context.clearRect( 0, 0, context.canvas.width, context.canvas.height );
 		context.fillText( set1, 0, 0 );
-		var rendered1 = canvas.toDataURL();
+		const rendered1 = context.getImageData(
+			0,
+			0,
+			context.canvas.width,
+			context.canvas.height
+		).data;
 
 		// Cleanup from previous test.
-		context.clearRect( 0, 0, canvas.width, canvas.height );
+		context.clearRect( 0, 0, context.canvas.width, context.canvas.height );
 		context.fillText( set2, 0, 0 );
-		var rendered2 = canvas.toDataURL();
+		const rendered2 = context.getImageData(
+			0,
+			0,
+			context.canvas.width,
+			context.canvas.height
+		).data;
 
-		return rendered1 === rendered2;
+		return rendered1.every( ( pixel, index ) => {
+			return pixel === rendered2[ index ];
```

Comment:
when I read this I was confused because I didn't know that the image data exposed pixels, and then I wondered what the data type of a pixel is, then I realized that actually no, these aren't pixels, these are the bytes that the pixels encode their RGBA color values into, and it's a `TypedArray`

maybe we can rename these so reflect that they are the separate 8-bit color channel values?

additionally, we might be able to create a `Uint32Array` view over these and compare faster, since each pixel in the image creates four bytes that we're individually comparing. I'm suggesting this because I assume the change from comparing the `toDataURL()` values is for performance? if we go that route, might as well pump the gas on it.

Reply:
> I'm suggesting this because I assume the change from comparing the `toDataURL()` values is for performance? if we go that route, might as well pump the gas on it.

Actually, the change is due to `toDataURL()` not being available in a Web Worker.

Reply:
@dmsnell How about 68879b0?

Reply:
looks fine to me!

> toDataURL() not being available in a Web Worker.

aha! interesting. I overlooked that.

I know we went back to `rendered1`, `rendered2`, and `rendered2Data`, which I think is fine. there's probably some wiggle room if we wanted to improve the names, but definitely not necessary for this patch.

ironically the `Uint32Array` is more like `pixels` than we had before.

```js
const pixels1 = new Uint32Array(…);
const pixels2 = new Uint32Array(…);

return pixels1.every( ( pixelBytes, index ) => pixelBytes === pixels2[ index ] );
```

not intending to over-suggest the name you had before or anything; just trying to follow your lead and see if now it makes more sense to indicate these are `bytes` or `pixels` or anything like that.

really love the `Uint32Array` - I _think_ that will further improve the efficiency, even though maybe at this time now it won't matter.